packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -1025,6 +1025,76 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
     return signatureWithModuleAddress;
   }
 
+  /**
+   * Deploys the smart contract
+   *
+   * This method will deploy a Smart Account contract. It is useful for deploying in a moment when you know that gas prices are low,
+   * and you want to deploy the account before sending the first user operation. This step can otherwise be skipped,
+   * as the deployment will alternatively be bundled with the first user operation.
+   *
+   * @param buildUseropDto {@link BuildUserOpOptions}.
+   * @returns Promise<{@link UserOpResponse}> that you can use to track the user operation.
+   * @error Throws an error if the account has already been deployed.
+   * @error Throws an error if the account has not enough native token balance to deploy, if not using a paymaster.
+   *
+   * @example
+   * import { createClient } from "viem"
+   * import { createSmartAccountClient } from "@biconomy/account"
+   * import { createWalletClient, http } from "viem";
+   * import { polygonMumbai } from "viem/chains";
+   *
+   * const signer = createWalletClient({
+   *   account,
+   *   chain: polygonMumbai,
+   *   transport: http(),
+   * });
+   *
+   * const smartAccount = await createSmartAccountClient({
+   *  signer,
+   *  biconomyPaymasterApiKey,
+   *  bundlerUrl
+   * });
+   *
+   * // If you want to use a paymaster...
+   * const { wait } = await smartAccount.deploy({
+   *   paymasterServiceData: { mode: PaymasterMode.SPONSORED },
+   * });
+   *
+   * // Or if you can't use a paymaster send native token to this address:
+   * const counterfactualAddress = await smartAccount.getAccountAddress();
+   *
+   * // Then deploy the account
+   * const { wait } = await smartAccount.deploy();
+   *
+   * const { success, receipt } = await wait();
+   *
+   */
+  public async deploy(buildUseropDto?: BuildUserOpOptions): Promise<UserOpResponse> {
+    const accountAddress = this.accountAddress ?? (await this.getAccountAddress());
+
+    // Check that the account has not already been deployed
+    const byteCode = await this.provider?.getBytecode({ address: accountAddress as Hex });
+    if (byteCode !== undefined) {
```

Comment:
have you tested this negative case?

Reply:
Yes, [here](https://github.com/bcnmy/biconomy-client-sdk/pull/429/files/8c3ebd512ee491806d7e79ce0c61bae71c2df621#diff-f9b56b02ea191c9a2bf7380a0dd2702bd35827ffefc83014b18bf2d03a904f76R568)